### DISCUSSION TITLE
How to compile C code that uses curl library correctly?

### DISCUSSION THREAD

**Discussion by mragochar** [curl/curl-for-win#41] [Q&A] [answered]
I have downloaded the library i.e. all  `.c`, `.a`, `.h` files and all other given prerequisites. All those are inside `C:/CURL` folder. I just try to give the path to `curl/curl.h` (as `#include<curl/curl.h>` is used in program) and the path to `libcurl.a` (for all those functions and anything related) to compile my code.

> gcc -IC:/CURL/include -LC:/CURL/lib  -lcurl test.c -o test

This generated a bunch of `undefined reference` error. How is this being so and can you guys help me to solve this issue of compilation.

**Comment by vszakats** [comment, marked as answer]
We had a recent discussion about static linking (though dynamic linking to libcurl DLL should be (much) simpler): https://github.com/curl/curl-for-win/issues/39

The linker command of the curl tool as it appears in the public build log may also serve as an example:
https://ci.appveyor.com/project/curlorg/curl-for-win/build/1.0.1813/job/0g06bdnijigxuw34?fullLog=true#L10450

Exact details will depend on compiler / toolchain / env / version / etc. So, not a comprehensive answer, but I hope it helps.

**Reply by mragochar** [reply]
Being a beginner, I stuck in every line and too hard to catch it. Can you please help me in simpler way?(if there is any)
